index.d.ts

```diff
@@ -0,0 +1,63 @@
+declare module "moleculer-apollo-server" {
+	import { ServiceSchema } from "moleculer";
+	import { Config } from "apollo-server-core";
+	export {
+		GraphQLUpload,
+		GraphQLExtension,
+		gql,
+		ApolloError,
+		toApolloError,
+		SyntaxError,
+		ValidationError,
+		AuthenticationError,
+		ForbiddenError,
+		UserInputError,
+		defaultPlaygroundOptions,
+	} from "apollo-server-core";
+
+	export * from "graphql-tools";
+
+	export interface ApolloServerOptions {
+		path: string;
+		disableHealthCheck: boolean;
+		onHealthCheck: () => {};
+	}
+
+	export class ApolloServer {
+		createGraphQLServerOptions(req: any, res: any): Promise<any>;
+		createHandler(options: ApolloServerOptions): void;
+		supportsUploads(): boolean;
+		supportsSubscriptions(): boolean;
+	}
+
+	export interface GraphQLNodeResolver {
+		action: string;
+		rootParams?: {
+			[key: string]: string;
+		};
+		dataloader?: boolean;
+	}
+
+	export interface GraphQLTypeResolver {
+		[key: string]: {
+			[key: string]: GraphQLNodeResolver;
+		};
+	}
+
+	export interface ApolloServiceOptions {
+		typeDefs?: string;
+		resolvers?: GraphQLTypeResolver;
+		routeOptions: {
```

Comment:
This doesn't cover all of the possible route options.  Unfortunately, the types for [moleculer-web](https://github.com/moleculerjs/moleculer-web) don't have them all defined either.  I'm not sure what the total set of possible options here is, but I'm getting an error because `authentication` is not defined as a property here.

Reply:
You're right. Currently. I don't know exactly the props should have. So, I could define like this:

```ts
  export interface ServiceRouteOptions {
		path: string;
		use?: any[];
		etag?: boolean;
		whitelist?: string[];
		authorization?: boolean;
		camelCaseNames?: boolean;
		aliases?: {
			[key: string]: any; // Should discuss more on this. string | AliasSchema, ...
		};
		bodyParsers?: {
			json: boolean;
			urlencoded: OptionsUrlencoded;
		};
		cors?: boolean | { origin: string[]; methods: string[] };
		callOptions?: {
			timeout: number;
			fallbackResponse?: any;
		};
		onBeforeCall: (ctx: Context, route: any, req: any, res: any) => Promise<any>;
		onAfterCall: (ctx: Context, route: any, req: any, res: any) => Promise<any>;
	}
```